app/controllers/time_slot_controller.rb

```diff
@@ -53,40 +52,24 @@ def cancel
   end
 
   def index
-    @appointment = current_patient.current_appointment
-    @ubs = @appointment.try(:ubs)
-
+    @appointment = current_patient.appointments.future.current
```

Comment:
This with `future.current` will ensure we only return @appointment if it's a future one.